x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/AutoBucketTests.java

```diff
@@ -84,15 +84,22 @@ private Expression build(Source source, Expression arg) {
         Literal from;
         Literal to;
         if (arg.dataType() == DataTypes.DATETIME) {
-            from = new Literal(Source.EMPTY, new BytesRef("2023-02-01T00:00:00.00Z"), DataTypes.KEYWORD);
-            to = new Literal(Source.EMPTY, new BytesRef("2023-03-01T00:00:00.00Z"), DataTypes.KEYWORD);
+            from = fromOrTo("2023-02-01T00:00:00.00Z");
+            to = fromOrTo("2023-03-01T09:00:00.00Z");
         } else {
             from = new Literal(Source.EMPTY, 0, DataTypes.DOUBLE);
             to = new Literal(Source.EMPTY, 1000, DataTypes.DOUBLE);
         }
         return new AutoBucket(source, arg, new Literal(Source.EMPTY, 50, DataTypes.INTEGER), from, to);
     }
 
+    private Literal fromOrTo(String date) {
+        if (randomBoolean()) {
```

Comment:
Nit, style preference: opportunity to use also here the ternary operator. But might break the lines making it harder to read.